code/game/objects/structures/mineral_doors.dm

```diff
@@ -852,7 +864,8 @@
 	repair_skill = /datum/skill/craft/masonry
 
 /obj/structure/mineral_door/wood/donjon/stone/attack_right(mob/user)
-	return
+	if(user.get_active_held_item())
+		..()
```

Comment:
Just remove this proc entirely, it's already handled by parent /obj/structure/mineral_door/wood/donjon/attack_right(mob/user)